source/apps/crawl_space/models.py

```diff
@@ -69,23 +70,15 @@ def get_absolute_url(self):
         return reverse('base:project',
             kwargs=dict(project_slug=self.project.slug))
 
-
     def save(self, *args, **kwargs):
-
         if self.pk is None:
-            super(CrawlModel, self).save(*args, **kwargs)
-
+            self.slug = slugify(self.name)
+            # TODO:
+            # Another weird call with a side effect that has to be fixed.
             model_path = self.ensure_model_path()
```

Comment:
You're right. I think the "weirdness" comes from the way the side effect is done, not that it has a side effect. I usually try to stay away from calling functions in other functions just for the purpose of causing a side effect, i.e.: 
```
ensure_exists(crawl_path)
```
Not how I would have done it.